### DISCUSSION TITLE
Steps To a Succesful Robot. From Basic Functionality to MADTOWN!??!??!?!!!!

### DISCUSSION THREAD

**Discussion by RudyG252** [Patribots4738/Crescendo2024#156] [General]
With the bot working and the shooter working and everything (kinda), the most important thing for the rest of the season is what we prioritize and what we can do to make us better than other bots, so I'm going to try to start a basic outline of our priorities. Any input is appreciated, as this is a rough draft and like a discussions and stuff!!! Keep in mind that when I'm making this list I'm under the assumption that we will have vision working at some point (Jacob's locked in!!!!)

Basic Functionality Priority List:
- [x] Shooting Speaker consistently (7 notes in a row should be consistency test) from a specific spot in the field to speaker.
- [x] auto align and shoot/place amp quickly
- [x] Climb without falling from left, right, and center (we can already do this!)
- [x] Preload + 1-2 piece auto (shooting from our set point)
- [x] Amp Consistently (same consistency test)
- [x] Have full control over the piece as it moves through the robot and be able to know where it is (by either accurately guessing via motor applied outputs or by indexing it in such a way that we can guarantee its exact position
- [ ] 2 piece amp auto (these are the basic autos any bot should be able to do IMO)

This list will probably get us to the point of an average bot at Ventura most likely, but we wanna be better than that so:

Better Than Basic Functionality (everything needs to pass consistency test):
- [x] Climb from any spot on the chain (chain mapping) (should already work realistically)
- [x] Tune interpolation map based on our math to make it more precise (pass seven piece consistency test close range)
- [x] Shoot somewhat consistently from close range (between stage and speaker using current arctan mode and full speeds) 
- [x] Shoot 2 pieces into speaker in quick succession
- [x] All wing pieces auto
- [x] Shooting from outside of short range (wing line)
- [ ] Auto limit testing (Assuming all our shooting is good news)
- [ ] Trap placement (not possible until elevator added)
- [x] Shooting while driving in short range (if this doesn't work it doesn't work we shouldn't stress, but I have high hopes)
- [ ] Intake from source using trapper

That's a draft, but any improvements would be great, or any reordering because I probably missed some stuff. I just want us to have a basic plan for what we want to be doing for the next couple of weeks. One thing I will mention is I want to make sure we don't overfocus on the speaker and that we give some love to amp. If we think from a strategic standpoint of the game, an alliance is useless without a good amp scorer, because no matter how fast you can cycle the speaker, it doesn't matter if you don't have it amped, so I want us to have the ability to score amp as quickly as possible. An alliance is only as strong as its weakest link, if that link is the one scoring amp!!!!!!

I also want everyone to think about what we can add to make us better than other teams. What can we add, that we have the ability to do, that can give us an edge? What are teams like madtown and hightide and citrus gonna do that's going to make them better than other bots? IMO in this game, CONSISTENCY IS KEY. The most important thing is that we never miss a piece, even if that means shooting from the same spot in teleop the whole match (which would suck). If you miss a piece right before amplification ends, it's not like charged up where it'll fall into hybrid or you'll just pick it back up again. You LOSE those points (4), because that piece is essentially being turned from an amped speaker piece (5 points) into your next amp piece (1 point), because you don't want to score it back in speaker as undamped. That's also not including the time you missed shooting and picking up that piece, so keep that in mind as well. That's my rant, give any input or any thoughts or anything you think of :D. 
// Working btw

**Comment by GalexY727** [comment]
Thank you for taking the time to write this, I think it will help a ton
